### PR TITLE
#67 Test for fetching binary content

### PR DESCRIPTION
**Pull request by carlosmiranda**

### CODE REVIEW

**Comment by karato** [comment]
Thanks, I will find someone to review this PR soon

**Comment by karato** [comment]
@darkled do this review

**Comment by asaen** [comment]
@carlosmiranda please resolve merge conflicts

**Comment by carlosmiranda** [comment]
@darkled done, please check again.

**Comment by asaen** [comment]
@carlosmiranda there is a PMD violation, could you please fix it as well, e.g. adding `@SuppressWarnings`?

```
[ERROR] /src/test/java/com/nerodesk/AppTest.java[76-296]: This class has too many methods, consider refactoring it. (TooManyMethods)
```

**Comment by carlosmiranda** [comment]
@darkled I pushed a new commit, please take another look.

**Comment by asaen** [comment]
@carlosmiranda thank you

**Comment by asaen** [comment]
@rultor merge

File: src/test/java/com/nerodesk/AppTest.java [hunk 1]
```diff
@@ -139,6 +139,36 @@ public void exec(final URI home) throws IOException {
...
 
     /**
+     * Application can return file content in binary form.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void returnsBinaryContent() throws Exception {
+        final Base base = new MkBase();
+        final String name = "test.dat";
+        final byte[] content = new byte[]{'t', 'e', 's', 't'};
```
Comment (yegor256) [inline src/test/java/com/nerodesk/AppTest.java]: I would suggest to use something less obvious here. Something like `{0x00, 0x0a, 0xff, 0xfe}`

**Comment by yegor256** [comment]
@carlosmiranda I think that this test is not actually reproducing a potential problem, see my comment above

File: src/test/java/com/nerodesk/AppTest.java (hunk 1 above)
Comment (carlosmiranda) [inline src/test/java/com/nerodesk/AppTest.java]: Makes sense, I'll do that.

**Comment by carlosmiranda** [comment]
@yegor256 I updated the test based on your review comments. Note that the casts on the latter two elements are due to the compiler complaining (something to do with the fact that Java has no unsigned types).

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by yegor256** [comment]
@rultor try to merge again

**Comment by asaen** [comment]
@carlosmiranda it seems there are some merge conflicts again...

**Comment by carlosmiranda** [comment]
@darkled @yegor256 Please try again. `App` was moved to `TsApp` hence the conflict.

**Comment by yegor256** [comment]
@rultor merge

**Comment by carlosmiranda** [comment]
@yegor256 Please try again, the checkstyle warnings were fixed.

**Comment by yegor256** [comment]
@rultor merge

**Comment by carlosmiranda** [comment]
@yegor256 Sorry, should be fixed now, please try again.

**Comment by yegor256** [comment]
@rultor merge

**Comment by karato** [comment]
@darkled Thank you, I have added **20 mins** to your account in payment/transaction "`54224017`", time consumed: 62 hours and 49 mins. extra minutes for review comments (c=5). added +20 to your rating, now it is equal to [+2244](http://www.netbout.com/b/35091?open=rating)

**Comment by karato** [comment]
@rultor deploy pls

**Pull request merged**